src/components/ProposalApplication.tsx

```diff
@@ -66,11 +77,58 @@ export default function ProposalApplication({
                     label: 'Details',
                     accessor: 'details',
                     transformer: ({ row }) => (
-                      <ApplicationDetailsModal row={row} />
+                      <ApplicationDetailsModal
+                        row={row}
+                        isModalOpen={isModalOpen}
+                        setIsModalOpen={setIsModalOpen}
+                      />
+                    ),
+                  },
+                  {
+                    label: 'Action',
+                    accessor: 'action',
+                    transformer: ({ row }) => (
+                      <Button
+                        disabled={
+                          row.statusKey !== 'OPEN' ||
+                          acceptApplication.isLoading
+                        }
+                        onClick={async () => {
+                          await acceptApplication.mutateAsync(
+                            {
+                              proposalId: proposalDetails.id,
+                              proposalApplicationId: row.id,
+                              applicantEmail: row.email,
+                            },
+                            {
+                              onSuccess: () => {
+                                refetch()
```

Comment:
we should change the filter to "MY_PROPOSALS" before refetching, as accepting an application on a proposal while filtering for, e.g., OPEN PROPOSALS will remove the proposal from the view after refetching (as it is then MATCHED and no longer OPEN). this is a bit confusing for the user. it would be nice if one can still see the proposal and the updated applications table

Reply:
Done (after refetching). Before would lead to empty page (MY_Proposals) & then refresh again.